### DISCUSSION TITLE
How can I get code blocks to wrap (in view mode)?

### DISCUSSION THREAD

**Discussion by mendhak** [gsantner/markor#2141] [Q&A] [answered]
Hi is it possible to get code blocks to wrap, so it fits the screen width? Text wrap works fine for normal text, just not the code blocks. 

In this example I've pasted some sample code between three backticks, and you can see the lines go past the edge. I'm just hoping to see all the code without having to horizontally scroll/drag.

![Screenshot_20231028-173349](https://github.com/gsantner/markor/assets/746276/1ac5e4b6-2973-48b7-9ec3-7180e501bbf2)

Thanks!

**Comment by gsantner** [comment, marked as answer]
You can use CSS and Markor's inject-to-head settings option for this.  The element that needs to be styled is `pre` or `code`. 

Documentation: <https://developer.mozilla.org/en-US/docs/Web/CSS/white-space>

**Reply by mendhak** [reply]
Thanks here is what I have added

```
pre {  word-wrap: break-word;
  white-space: pre-wrap;
}
code {  word-wrap: break-word;
  white-space: pre-wrap;
}
```

**Comment by guanglinn** [comment]
I implemented the feature to wrap words in view-mode code blocks, see PR #2129.
